challenges/frontend/src/components/ContactForm.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { GoogleReCaptchaProvider, GoogleReCaptcha } from 'react-google-recaptcha-v3';
+
+class ContactForm extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.state = {
+      name: '',               // Name filled into contact form
+      email: '',              // Email filled into contact form
+      messageBody: '',        // Message filled into contact form
+      recaptchaValid: false,  // Set to true in the verifyCaptcha callback
+      isFormValid: false      // Determines if submit button will be enabled
+    };
+    
+    this.handleFormChange = this.handleFormChange.bind(this);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.validateForm = this.validateForm.bind(this);
+    this.verifyCaptcha = this.verifyCaptcha.bind(this);
+  }
+
+  // Checks whether the values for a given state
+  // are in a format that can be submitted.
+  // For now we just check that they are not empty
+  // and that the captcha is valid.
+  validateForm(state) {
+    return state.name !== "" 
+      && state.email !== "" 
+      && state.messageBody !== ""
+      && state.recaptchaValid;
+  }
+  
+  // Called when a value from a form control changes.
+  // We check if the form is valid and update the
+  // state accordingly.
+  handleFormChange(event) {
+    var state = this.state;
+    state[event.target.name] = event.target.value
+    
+    this.setState({
+      [event.target.name]: event.target.value, 
+      isFormValid: this.validateForm(state)
+    });
+  }
+  
+  // Called when the submit button is pressed. Logs
+  // contact information into the console.
+  handleFormSubmit(event) {
+    event.preventDefault();
```

Comment:
I had to add this line as the default behavior for submitting a form seemed to be reloading the page. This lets me have my control.